### ISSUE TITLE
Possible bug when deleting: Not fully deleting?

### ISSUE THREAD

**Issue by eklem** [fergiemcdowall/search-index#514]
Trying to rewrite the examples and found a possible bug when adding - searching - deleting - searching one more time. The document is still semi-available with `db.QUERY({ DOCUMENTS: ['3'] })`.

You can clone and do a `git switch fix-examples` and then check [out docs/examples/DELETE.js](https://github.com/fergiemcdowall/search-index/blob/fix-examples/docs/examples/DELETE.js).

Document count and show all documents seems correct
```javaScript
await db.DOCUMENT_COUNT().then(console.log)
await db.ALL_DOCUMENTS().then(console.log)
 ```

**Comment by fergiemcdowall** [comment]
yes- this is not working here either- taking a look...

**Comment by fergiemcdowall** [comment]
There were a couple of bugs here and a bit of bad documentation. I have fixed the bugs.

There is a `PUT` option that is called `storeVectors`. This has to be `true` for the index to be deletable. I have set the default to be false, since it makes indexes smaller. Maybe it should be `true` by default?

Either way, the examples in `docs/examples` still need to be updated, so I will have a look at them ASAP.

**Comment by eklem** [comment]
Nice! Think it should be set to true for two reasons:
* It limits functionality and can be hard to figure out when starting out testing search-index
* Differentiate from Lunr?

**Comment by eklem** [comment]
I'll do a pull and check the last examples. Possibly one I can't figure out, bit the rest should soon be good.

**Comment by eklem** [comment]
Now things are looking better! But one question.

When doing:
```javaScript
  console.log('\nDELETING ->')
  await db.DELETE(['3']).then(console.log)

  console.log('\nCONFIRM DELETE ->')
  await db.QUERY({ DOCUMENTS: ['3'] }).then(console.log)
  await db.DOCUMENT_COUNT().then(console.log)
```

Why is result_length when querying still 1?
```console
DELETING ->
[ { _id: '3', operation: 'DELETE', status: 'DELETED' } ]

CONFIRM DELETE ->
{ RESULT: [ null ], RESULT_LENGTH: 1 }
2
```

**Comment by fergiemcdowall** [comment]
Because `[ null ].length == 1` and `[ null, null ].length == 2`

You might ask for `db.QUERY({ DOCUMENTS: ['4', '11', '2', '3'] })` and get the result `[ null, { /*...*/ }, null, null ]`, and that way you can see at a glance that '11' is the only document that was found

That said, as I am explaining it, I can see how it might be neater to only return the DOCUMENTS that are found, and strip out all of the `null`s. That would also mean that `RESULT_LENGTH` would make more sense.

Maybe the expected result above should be `{ RESULT: [ null ], RESULT_LENGTH: 1 }`?

**Comment by eklem** [comment]
I guess I'm a little bit too much thinking regular search results. Here, you probe for one or more specific documents. Then the count isn't so important, but the null's are. Got really unsure, but I think you should leave it as it is for now.

**Comment by mew-york** [comment]
Hi,

Related to this issue, I made a minimal adaptation of the DELETE example from https://replit.com/@fergiemcdowall/search-index-DELETE#index.js

I'm surprised that I still get results for searching "Zeppelin" after the document in which it appears is deleted from the db.  Here's the code and below are the console messages that get printed. I guess I was expecting to see
CONFIRM DELETE ->
{ RESULT: [], RESULT_LENGTH: 0 }
if I search for something that is no longer in the db.

Thanks and interested to hear your thoughts!

```
(async () => {
  const si = require('search-index')
  const db = await si({
    name: 'TMP-DELETE',
    storeVectors: true
  })

  console.log('\nADDING ->')
  await db.PUT([
    {
      _id: 1,
      bandName: 'The Beatles',
      description: 'The nice boys of pop'
    }, {
      _id: 'two',
      bandName: 'The Rolling Stones',
      description: 'The bad boys of rock'
    }, {
      _id: 3,
      bandName: 'The Who',
      description: 'Nearly as good as Led Zeppelin'
    }
  ], { storeVectors: true }).then(console.log)

  console.log('\nVERIFYING ->')
  await db.SEARCH(["Zeppelin"]).then(console.log)

  console.log('\nDELETING ->')
  await db.DELETE(3).then(console.log)

  console.log('\nCONFIRM DELETE ->')
  await db.SEARCH(["Zeppelin"]).then(console.log)
})()
```

ADDING ->
[
  { _id: 1, operation: 'PUT', status: 'CREATED' },
  { _id: 'two', operation: 'PUT', status: 'CREATED' },
  { _id: 3, operation: 'PUT', status: 'CREATED' }
]

VERIFYING ->
{
  RESULT: [ { _id: 3, _match: [Array], _score: 0.69 } ],
  RESULT_LENGTH: 1
}

DELETING ->
[ { _id: 3, operation: 'DELETE', status: 'DELETED' } ]

CONFIRM DELETE ->
{
  RESULT: [ { _id: 3, _match: [Array], _score: 0.55 } ],
  RESULT_LENGTH: 1
}

**Comment by fergiemcdowall** [comment]
Hmm- thats odd- yes there does seem to be a bug here. Will try and take a proper look at it on monday.

**Comment by fergiemcdowall** [comment]
Thanks for the bug report @mew-york - this is now fixed in `search-index@3.4.0`

**Comment by mew-york** [comment]
Thank you for this, Fergus!
